File: app/src/main/java/com/zskjprojectj/andouclient/utils/PayUtil.java
package com.zskjprojectj.andouclient.utils;

import android.content.Context;

import com.tencent.mm.opensdk.modelpay.PayReq;
import com.tencent.mm.opensdk.openapi.IWXAPI;
import com.tencent.mm.opensdk.openapi.WXAPIFactory;
import com.zskjprojectj.andouclient.entity.WXPayBean;

public class PayUtil {
    public static void startWXPay(Context context, WXPayBean wxPayBean) {
        final IWXAPI msgApi = WXAPIFactory.createWXAPI(context, wxPayBean.getAppid());
        msgApi.registerApp(wxPayBean.getAppid());
        PayReq req = new PayReq();
        req.appId = wxPayBean.getAppid();
        req.partnerId = wxPayBean.getMch_id();
        req.prepayId = wxPayBean.getPrepay_id();
        req.packageValue = "Sign=WXPay";
        req.nonceStr = wxPayBean.getNonce_str();
        req.timeStamp = wxPayBean.getTimestamp();
        req.sign = wxPayBean.getSign();
        msgApi.sendReq(req);
    }
}

File: app/src/main/java/com/zskjprojectj/andouclient/utils/PayUtil.kt
package com.zskjprojectj.andouclient.utils

import android.content.Context
import com.blankj.utilcode.util.ToastUtils
import com.tencent.mm.opensdk.modelpay.PayReq
import com.tencent.mm.opensdk.openapi.WXAPIFactory
import com.zskjprojectj.andouclient.model.WxPay

object PayUtil {
    fun startWXPay(context: Context, wxPayBean: WxPay) {
        val msgApi = WXAPIFactory.createWXAPI(context, WEI_XIN_APP_ID)
        msgApi.registerApp(WEI_XIN_APP_ID)
        if (!msgApi.isWXAppInstalled) {
            ToastUtils.showLong("您还没有安装微信")
            return
        }
        val req = PayReq()
        req.appId = WEI_XIN_APP_ID
        req.partnerId = wxPayBean.partnerid
        req.prepayId = wxPayBean.prepayid
        req.packageValue = "Sign=WXPay"
        req.nonceStr = wxPayBean.noncestr
        req.timeStamp = wxPayBean.timestamp
        req.sign = wxPayBean.sign
        msgApi.sendReq(req)
    }
}